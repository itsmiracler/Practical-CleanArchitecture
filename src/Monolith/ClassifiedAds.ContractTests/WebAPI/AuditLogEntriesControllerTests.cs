using ApprovalTests;
using ApprovalTests.Reporters;
using ClassifiedAds.Application.AuditLogEntries.DTOs;
using NJsonSchema;
using Xunit;

namespace ClassifiedAds.ContractTests.WebAPI
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class AuditLogEntriesControllerTests
    {
        [Fact]
        public void AuditLogEntryDTOTest()
        {
            var schema = JsonSchema.FromType<AuditLogEntryDTO>().ToJson();
            Approvals.Verify(schema);
        }
    }
}
