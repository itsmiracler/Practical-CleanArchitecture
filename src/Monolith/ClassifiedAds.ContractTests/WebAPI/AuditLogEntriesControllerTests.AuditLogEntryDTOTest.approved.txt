{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "AuditLogEntryDTO",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "Id": {
      "type": "string",
      "format": "guid"
    },
    "UserId": {
      "type": "string",
      "format": "guid"
    },
    "UserName": {
      "type": [
        "null",
        "string"
      ]
    },
    "Action": {
      "type": [
        "null",
        "string"
      ]
    },
    "ObjectId": {
      "type": [
        "null",
        "string"
      ]
    },
    "Log": {
      "type": [
        "null",
        "string"
      ]
    },
    "CreatedDateTime": {
      "type": "string",
      "format": "date-time"
    }
  }
}